### PR TITLE
group api

### PR DESCRIPTION
**Pull request by sywhb**
- Add is_admin field to group_membership table
- Add createGroup API
- Add groups API

### CODE REVIEW

File: packages/api/src/resolvers/recommendations/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      return {
+        errorCodes: [CreateGroupErrorCode.Unauthorized],
+      }
+    }
+
+    const [group, invite] = await createGroup({
+      admin: userData,
+      name: input.name,
+      maxMembers: input.maxMembers,
+      expiresInDays: input.expiresInDays,
```
Comment (jacksonh) [inline packages/api/src/resolvers/recommendations/index.ts:42]: nice, what is the default for expiration?

File: packages/api/src/entity/groups/group.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { User } from '../user'
 @Entity()
 export class Group {
```
[the remark this answers is not in the document]
Comment (sywhb) [reply packages/api/src/entity/groups/group.ts:14]: Yes, in the `invite` table

File: packages/api/src/resolvers/recommendations/index.ts (hunk 1 above)
Comment (sywhb) [reply packages/api/src/resolvers/recommendations/index.ts:42]: 7 days

File: packages/api/src/entity/groups/group.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jacksonh) [reply packages/api/src/entity/groups/group.ts:14]: nice, thanks. Forgot we had that already.

File: packages/api/src/resolvers/recommendations/index.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    })
+
+    const inviteUrl = `${homePageURL()}/invite/${invite.code}`
```
Comment (jacksonh) [inline packages/api/src/resolvers/recommendations/index.ts:45]: 👍

File: packages/api/src/resolvers/recommendations/index.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      where: { id: uid },
+      relations: ['profile'],
+    })
+    if (!userData) {
+      return {
+        errorCodes: [CreateGroupErrorCode.Unauthorized],
+      }
+    }
+
+    const [group, invite] = await createGroup({
```
Comment (jacksonh) [inline packages/api/src/resolvers/recommendations/index.ts:38]: Is there a limit to the number of groups created per-user anywhere?

Comment (sywhb) [reply packages/api/src/resolvers/recommendations/index.ts:44]: Yes, just added here:
https://github.com/omnivore-app/omnivore/pull/1493/files#diff-91e988313cbb1cf4f86834197ca727146e7766377f232ee278bd4333bb0e2dd4R20-R28

**Review by jacksonh — APPROVED** [review]

**Pull request merged**